### PR TITLE
(svelte) - Reimplement @urql/svelte bindings with new API approach

### LINKED ISSUE #1007: RFC: Intent to implement stable Svelte bindings
## Summary

Currently I see our current `@urql/svelte` package and its past and current implementation as nice experiments which helped check the boundaries of Svelte and test APIs and the understanding of users.

The latest PR allowed _all_ modes of Svelte's stateful stores to be used somehow, with similar semantics to the other bindings, See #862 

It's becoming increasingly clear that this is confusing and doesn't feel idiomatic to Svelte.
The next implementation of `@urql/svelte` should start from a clean slate and be idiomatic to Svelte and intuitively avoid confusion. It is allowed to stray away from feeling like the `urql` and `@urql/preact` bindings, even forgoing permanent state memory if necessary.

## Problems with the approach

The current approach attempts to allow three modes of usage:

- Store usage without memoised state
- Store usage with memoised state via an intermediate factory function
- Async-await/Promise usage

The last option of these feels idiomatic to Svelte but _should be discouraged_. We never aim to show results that can never reactively update. This is objectively bad.

The memoised usage is confusing and Svelte simply does not have a full reactivity model that allows us to observe changes. The user can do this manually and this may feel wrong to us, trying to implement complex hooks flows in Svelte, however this is not our problem to solve.

## Bugs to avoid

- Unmounting issues, which may be inherent to our implementation: #948, #979
- Incorrect promise behaviour #932

### PR DESCRIPTION
**Pull request by kitten**
Resolve #1007 

This is intended to replace the current Svelte bindings as either a new breaking beta release, or as the stable v1 release.

The initial idea is to move towards an approach that fully encapsulates the `OperationResult` output and `GraphQLRequest` input states into one merged `Writable` store. This store can be observed by components and by operations, e.g. `query` and `subscription`.

```svelte
<script>
  import gql from 'graphql-tag';
  import { operationStore, query } from '@urql/svelte';

  // We initialise out query container, which represents both the query input and result
  const todos = operationStore(gql`
    query ($first: Int!, $skip: Int) {
      todos(first: $first, skip: $skip) {
        id
      }
    }
  `, { first: 10 });

  // We instruct urql to start the query and keep it updated
  // This is called at most once since it's tied to the component lifecycle
  query(todos);

  // We can reactively update the container
  const next = () => {
    $todos.variables.skip = 10;
  };
</script>

{#if $todos.fetching}
  <p>Fetching...</p>
{:else if $todos.error}
  <p>Error! {$todos.error.message}</p>
{:else}
  <ul>
    {#each $todos.data.todos as todo}
      <li>{todo.id}</li>
    {/each}
  </ul>
  <button on:click={next}>Next Page</button>
{/if}
```

This effectively and intuitively steers users away from pitfalls and covers easy to miss edge cases, where our previous implementation would create memory leaks. This API is focused on providing a single way to do things; this is especially advantageous since `client.query` shortcut methods and the likes are still accessible.

So overall this focuses on a single primitive to create a value container in Svelte that is only mutable in certain ways (only `query`, `variables`, and `context` can be changed). The other constraint is that operations like `query` and `subscription` should be called at most once, which is easy to teach and looks as imperative as most Svelte code does anyway.

Another nice advantage of the `OperationStore` is that it's both the stateful object and a Svelte writable store in one. It may still be passed around, outside of Svelte components, and can be mutated and read, since it provides getters and setters and prevents changes to result properties to keep things safe.

### CODE REVIEW

**Comment by frederikhors** [comment]
Amazing! Can't wait to try it!

Latest commit is broken.

**Comment by frederikhors** [comment]
Real-time reproduction with latest commit: https://codesandbox.io/s/urql-svelte-crud-with-new-bindings-rbrsb?file=/List.svelte.

Still the problem with navigation from `/` to `/todos` page.

If you reload on `/todos` page it works.

**Comment by kitten** [comment]
@frederikhors Yep, I think there's a small timing issue still where some synchronous retries lead to our Svelte implementation becoming stuck. But once we fix that this will be ready to go :+1:

**Comment by kitten** [comment]
It's looking pretty fine to me now; the Graphache timing fix for the `offlineExchange` has been merged to `main` and this branch has been rebased to keep it clean. https://codesandbox.io/s/urql-svelte-crud-pollinterval-keeps-calling-forked-lseuf?file=/urql.js

Since this is looking very promising I'd like to take some time to write some docs sections for Svelte and ship this branch as the first stable `@urql/svelte` v1.

File: packages/svelte-urql/src/operationStore.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
+import { Readable, writable } from 'svelte/store';
+import { DocumentNode } from 'graphql';
+import { OperationContext, CombinedError } from '@urql/core';
+
+import { _storeUpdate } from './internal';
+
+const noop = Object.create(null);
```
Comment (JoviDeCroock) [inline packages/svelte-urql/src/operationStore.ts:7]: Should we call this dict since `noop` refers to a func

**Review by JoviDeCroock — APPROVED** [review]
This looks awesome

**Comment by kitten** [comment]
This has just been published as `@urql/svelte@1.0.0`

**Pull request merged**